modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy

```diff
@@ -35,6 +35,27 @@ class LsfExecutor extends AbstractGridExecutor {
 
     private static char BLANK = ' ' as char
 
+    // parse the memory units from $LSF_ENVDIR/lsf.conf
+    def getLSF_memoryUnit() {
+        // assuming $LSF_ENVDIR/lsf.conf exists
+        def lsf_config = new File(System.getenv("$LSF_ENVDIR").toString() + "/lsf.conf")   // should write check to throw informative error if $$LSF_ENVDIR/lsf.conf not found
+        def lines = file1.readLines()
+        def found = lines.find{ line-> line =~ /LSF_UNIT_FOR_LIMITS/ }
+        def memory_units = found.split('LSF_UNIT_FOR_LIMITS=')[1]
+        return memory_units
+    }
+
+    /**
+     * Add an init method in the LsfExecutor overriding the base init() in AbstractGridExecutor.groovy
+    */
+    @Override
+    void init() {
+        super.init()
+        queueInterval = session.getQueueStatInterval(name)
+        log.debug "Creating executor '$name' > queue-stat-interval: ${queueInterval}"
+        def memory_lsf_unit_for_limits = getLSF_mermoryUnit()
+    }
```

Comment:
There aren't so much code convention, but surely NF uses camelCase instead of under_scores

Reply:
agreed!